lang/src/org/partiql/lang/planner/transforms/AstToLogicalVisitorTransform.kt

```diff
@@ -54,51 +54,46 @@ internal class AstToLogicalVisitorTransform(
             PartiqlLogical.build { limit(transformExpr(it), algebra, node.limit.metas) }
         } ?: algebra
 
-        return convertProjectionToBindingsToValues(node, algebra)
+        var expr = transformProjection(node, algebra)
+
+        // SELECT DISTINCT ...
+        expr = if (node.setq != null && node.setq is PartiqlAst.SetQuantifier.Distinct) {
+            return PartiqlLogical.build { call("filter_distinct", expr) }
+        } else expr
```

Comment:
This is a bit odd.

```suggestion
        expr = if (node.setq != null && node.setq is PartiqlAst.SetQuantifier.Distinct) {
            PartiqlLogical.build { call("filter_distinct", expr) }
        } else expr
```

OR

```suggestion
        if (node.setq != null && node.setq is PartiqlAst.SetQuantifier.Distinct) {
            return PartiqlLogical.build { call("filter_distinct", expr) }
        }
```

Reply:
I don't know what I was thinking hahaha thanks